src/yardstick/store/tool.py

```diff
@@ -1,10 +1,13 @@
 from __future__ import annotations
 
 import os
+from typing import TYPE_CHECKING
 
-from yardstick import artifact
 from yardstick.store import config as store_config
 
+if TYPE_CHECKING:
```

Comment:
I'm sure this is fine, but I don't understand why this guard is necessary. Why don't we always want to import `artifact`?

Reply:
this is a common pattern and is unique to python type hints (I haven't seen anything like this with other languages). Since importing takes time and these types are only used for type hints (and type hints are not used a runtime) then conditionally importing these based on the context (at runtime or only during type checking) makes sense for performance reasons. That being said, I'm not saying that there is a real performance cost here, but the linting tool is currently configured to find these optimizations and error out when conditinal type checking is not done (but could be) as a common convention.